### ISSUE TITLE
Simultanous fetch clamping required for Lightnovelbastion

### ISSUE THREAD

**Issue by nandakishore2009** [dteviot/WebToEpub#408]
Simultaneous fetch of 4 or more chapters results in the IP getting banned temporarily on Lightnovelbastion. 
![Capture](https://user-images.githubusercontent.com/24523188/98158575-202a3f00-1f01-11eb-816a-bf19fefdda77.JPG)
My suggestion is to clamp max fetches to 2(tested it working) requests.

**Comment by nandakishore2009** [comment]
@dteviot 
During this ordeal, a **premature epub save dialog popup** bug is also noticed, where the save dialog window popup premature followed by the progress bar and the chapter download status icon getting stuck unfinished. For error simulation:
URL: https://lightnovelbastion.com/novel/death-mage/
Max Fetch: 2
Used extension DotVPN set to Netherland

**Comment by dteviot** [comment]
@nandakishore2009 
> During this ordeal, a premature epub save dialog popup bug is also noticed, where the save dialog window popup premature

I assume you mean, WebToEpub stopped collecting chapters before it had collected all the chapters it was supposed to.
This is by design.  If it fails to fetch a chapter with an unrecoverable network error, it stops, saves what it has, and lets the user decide what to do.

> chapter download status icon getting stuck unfinished

That is a known bug.  Unfortunately the code to handle network errors is VERY complicated, and I've not been able to figure out how to fix that problem.  Basically, 
1. The problem itself isn't a big problem when encountered.
2. Fixing it is a lot of work, with a very high risk of breaking something else.
3. So, I've decided to not try spending any more time trying to fix it.

If you want, you're welcome to try and fix it.

**Comment by dteviot** [comment]
@nandakishore2009
I've added the rate limiting.  (Note, I've set max rate to 1 for Lightnovelbastion..)
Test versions for Firefox and Chrome have been uploaded to https://drive.google.com/drive/folders/1B_X2WcsaI_eg9yA-5bHJb8VeTZGKExl8?usp=sharing. Follow the "How to install from Source" instructions at https://github.com/dteviot/WebToEpub#how-to-install-from-source and let me know if that solves most of your problems.
